app/routes.py

```diff
@@ -0,0 +1,334 @@
+from app import db
+from app.models.customer import Customer
+from flask import Blueprint, jsonify, make_response, request
+from datetime import datetime, date, timedelta
+from app.models.video import Video
+import requests, json
+from dotenv import load_dotenv
+import os
+from app.models.utility_func import *
+from app.models.rental import Rental
+
+
+customer_bp = Blueprint("customer", __name__, url_prefix="/customers")
+video_bp = Blueprint("video", __name__, url_prefix="/videos")
+rental_bp = Blueprint("rental", __name__, url_prefix="/rentals")
+
+@customer_bp.route("", methods=["GET"])
+def get_all_customers():
+    all_customers = Customer.query.all()
+
+    if not all_customers:
+        return jsonify([]), 200
+    
+    customers_list = [] 
+    for customer in all_customers:
+        customer_dict = {
+            "id" : customer.id, 
+            "name" : customer.name, 
+            "postal_code" : customer.postal_code,
+            "phone": customer.phone,
+            "register_at" : customer.register_at
+        }
+        customers_list.append(customer_dict)
+    
+    return jsonify(customers_list), 200
+
+
+@customer_bp.route("/<customer_id>", methods=["GET"])
+def get_a_customer(customer_id):
+    if not customer_id.isnumeric():
+        return {'message': "Invalid type"}, 400
+
+    try:
+        customer = Customer.query.get(customer_id)
+
+        return  {
+        "id" : customer.id, 
+        "name" : customer.name,   
+        "postal_code" : customer.postal_code,
+        "phone": customer.phone,
+        "register_at" : customer.register_at
+        }, 200
+
+    except:
+        return {'message': f"Customer {customer_id} was not found"}, 404
+
+@customer_bp.route("", methods=["POST"])
+def post_a_customer():
+    request_body = request.get_json()
+
+    try:
+        new_customer = Customer(
+            name = request_body["name"],
+            postal_code = request_body["postal_code"],
+            phone = request_body["phone"],
+            register_at = datetime.now()
+            )
+        db.session.add(new_customer)
+        db.session.commit()
+
+        return {"id" : new_customer.id}, 201
+    except KeyError as err:
+        if "name" in err.args:
+            return {"details" : f"Request body must include name."}, 400
+        if "postal_code" in err.args:
+            return {"details" : f"Request body must include postal_code."}, 400
+        if "phone" in err.args:
+            return {"details" : f"Request body must include phone."}, 400
```

Comment:
you could create a function to handle these guard clauses dynamically and then use that function for customers and videos it could look something like
```
def validate_parameters(request_body, parameters):
           for attribute in parameters:
                if attribute not in request_body:
                       abort(make_response({"details": f"Request body must include {attribute}."}, 400))

```